taca_ngi_pipeline/deliver/deliver_dds.py

```diff
@@ -0,0 +1,546 @@
+        stage_dir = self.expand_path(self.stagingpath)
+        log_dir = os.path.join(os.path.basename(CONFIG.get('log').get('file')), 'DDS_logs') 
+        project_log_dir = os.path.join(log_dir, self.projectid)
+        cmd = ['dds', 'data', 'put', 
+               '--mount-dir', project_log_dir,
+               '--project', name_of_delivery, 
+               '--source', stage_dir]
+        try:
+            output = subprocess.check_output(cmd).decode('utf-8')
+        except subprocess.CalledProcessError as e:
+            logger.exception('DDS upload failed while uploading {} to {}'.format(stage_dir, name_of_delivery))
+        if "Upload completed!" in output:
+            delivery_status = "uploaded"
+        return delivery_status
+
+    def get_samples_from_charon(self, delivery_status='STAGED'):
+        """Takes as input a delivery status and return all samples with that delivery status
+        """
+        charon_session = CharonSession()
+        result = charon_session.project_get_samples(self.projectid)
+        samples = result.get('samples')
+        if samples is None:
+            raise AssertionError('CharonSession returned no results for project {}'.format(self.projectid))
+        samples_of_interest = []
+        for sample in samples:
+            sample_id = sample.get('sampleid')
+            charon_delivery_status = sample.get('delivery_status')
+            if charon_delivery_status == delivery_status or delivery_status is None:
+                samples_of_interest.append(sample_id)
+        return samples_of_interest
+
+    def _create_delivery_project(self):
+        """Create a DDS delivery project and return the ID
+        """
+        create_project_cmd = ['dds', 'project', 'create',
+                              '--title', self.project_title,
+                              '--description', self.project_desc,
+                              '--principal-investigator', self.pi_name,
+                              '--owner', self.pi_email]
+        if self.other_member_details:
+            for member in self.other_member_details:
+                create_project_cmd.append('--researcher')
+                create_project_cmd.append(member)
+        if self.sensitive:
+            create_project_cmd.append('--is_sensitive')
+        dds_project_id = ''
+        try:
+            output = subprocess.check_output(create_project_cmd, stderr=subprocess.STDOUT).decode("utf-8") #TODO: get more output - test this
+            project_pattern = re.compile('ngis\d{5}')  #TODO: print more info to the log (like "User sarasjunnebo was associated with Project ngis00043 as Owner=True. An e-mail notification has not been sent")
+            dds_project_id = re.search(project_pattern, output).group()
+            logger.info("DDS project successfully set up for {}. Info:\n".format(self.projectid, output)) #TODO: output is not printed
+        except Exception as e:  #TODO: explicitly handle when new user is added (exit 2)
```

Comment:
Not sure if the cli would return an Exception. We should check stderr for errors too, if the project hasn't been created.

Reply:
Good point, I'll have a look 👍